src/plugins/EnergyRAPL.cpp

```diff
@@ -0,0 +1,154 @@
+	MPI_Comm_rank(MPI_COMM_WORLD, &_thisRank);
+#endif
+	std::ostringstream outputFilename;
+	outputFilename << _outputprefix << ".tsv";
+	std::ofstream outputFile(outputFilename.str().c_str());
+	outputFile << "milliseconds\tsimstep\tjoules" << std::endl;
+	// For each package...
+	int maxPackageIdx = 0;
+	int numberOfPackages = getNumberOfPackages();
+	for (int packageIdx = 0; packageIdx < numberOfPackages; packageIdx++) {
+		std::ostringstream packageBasePath;
+		packageBasePath << _basePathRAPL << "intel-rapl:" << packageIdx;
+		// add package domain and...
+		Log::global_log->info() << "[" << getPluginName() << "] Adding package domain " << packageBasePath.str()
+								<< std::endl;
+		_counters.push_back(RAPLCounter(packageBasePath.str()));
+		// for each domain in package...
+		for (int domainIdx = 0;; domainIdx++) {
+			std::ostringstream domainBasePath;
+			domainBasePath << packageBasePath.str() << "/intel-rapl:" << packageIdx << ":" << domainIdx;
+			std::ostringstream domainNamePath;
+			domainNamePath << domainBasePath.str() << "/name";
+			std::string domainName;
+			std::ifstream domainNameFile(domainNamePath.str());
+			if (!domainNameFile.good()) break;  // (if domain exists...)
+			domainNameFile >> domainName;
+			if (0 != domainName.compare("dram")) continue;  // (and is DRAM domain)
+			// add domain
+			Log::global_log->info() << "[" << getPluginName() << "] Adding DRAM domain " << packageBasePath.str()
+									<< std::endl;
+			_counters.push_back(RAPLCounter(packageBasePath.str()));
+		}
+	}
+	_simstart = std::chrono::steady_clock::now();
+	for (auto counter : _counters) counter.reset();
+}
+
+void EnergyRAPL::endStep(ParticleContainer* particleContainer, DomainDecompBase* domainDecomp, Domain* domain,
+						 unsigned long simstep) {
+	_joules = 0;
+	for (auto counter : _counters) _joules += counter.update();
+	_simstep = simstep;
+	if (0 < _writeFrequency && simstep % _writeFrequency == 0) outputEnergyJoules();
+}
+
+void EnergyRAPL::readXML(XMLfileUnits& xmlconfig) {
+	xmlconfig.getNodeValue("writefrequency", _writeFrequency);
+	xmlconfig.getNodeValue("outputprefix", _outputprefix);
+}
+
+void EnergyRAPL::outputEnergyJoules() {
+	double joules = _joules;
+#ifdef ENABLE_MPI
+	// Collect results from all nodes (matching of separate messages over tag)
+	if (_thisRank == 0) {
+		std::map<std::string, double> nodeJoules;
+		nodeJoules[_processorName] = joules;  // Store result of rank 0
+		int numberOfRanks;
+		MPI_Comm_size(MPI_COMM_WORLD, &numberOfRanks);
+		for (int otherRank = 1; otherRank < numberOfRanks; otherRank++) {
+			MPI_Status status;
+			MPI_Recv(_processorName, MPI_MAX_PROCESSOR_NAME, MPI_CHAR, otherRank, /* tag */ otherRank, MPI_COMM_WORLD,
+					 &status);
+			MPI_Recv(&joules, 1, MPI_DOUBLE, otherRank, /* tag */ otherRank, MPI_COMM_WORLD, &status);
+			nodeJoules[_processorName] = joules;
+		}
+		joules = 0;
+		// TODO maybe output the energy consumption per node
```

Comment:
either fix the todo or open an issue and link it here

Reply:
This was more of a _hint_ at where this should be implemented if it is ever needed. (It is not needed right now.)
I have to admit, it's not good to add random To-dos to the code, so I'll remove it.